### PR TITLE
refactor: adding lovely-pytest-docker code

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
This PR contains following changes

- Added code for lovely-pytest-docker in PSA and  migrated to v2 version of docker-compose as GitHub runners have stopped supporting docker compose v1.
- Also fixed the CI runs for e2e tests as it always running on latest splunk instead of version provided by addonfactory-splunk-matrix.

### CODE REVIEW

File: pytest_splunk_addon/docker_class.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
+#
+# Copyright 2021 Splunk Inc.
```
Comment (harshilgajera-crest) [inline pytest_splunk_addon/docker_class.py:2]: @artemrys I have added this copyright license. Hope that is fine.

Comment (artemrys) [reply pytest_splunk_addon/docker_class.py:2]: Actually I am not sure - I'll ask our legal person if that's okay.

Please put it as an exception for now to `.licenserc.yaml`.

File: .github/workflows/build-test-release.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
       - id: matrix
-        uses: splunk/addonfactory-test-matrix-action@v1
```
Comment (artemrys) [inline .github/workflows/build-test-release.yml:24]: Let's revert this change as well. Let's stick to the latest possible version of Splunk.

Comment (harshilgajera-crest) [reply .github/workflows/build-test-release.yml:24]: Yes, will do that.

File: pytest_splunk_addon/docker_class.py (hunk 1 above)
Comment (harshilgajera-crest) [reply pytest_splunk_addon/docker_class.py:2]: Will do

Comment (harshilgajera-crest) [reply pytest_splunk_addon/docker_class.py]: Done

File: .github/workflows/build-test-release.yml (hunk 2 above)
Comment (harshilgajera-crest) [reply .github/workflows/build-test-release.yml]: Done

**Review by artemrys — APPROVED** [review]
Good job!

**Comment by dkvashninsplunk** [comment]
This breaks ALL tests, due to knowledge tests dependency on lovely-pytest-docker.

https://github.com/lovelysystems/lovely-pytest-docker/blob/dc87ba3397223305db16a542cce123110ac93864/src/lovely/pytest/docker/compose.py#L212C1-L218C73

`https://github.com/lovelysystems/lovely-pytest-docker/blob/dc87ba3397223305db16a542cce123110ac93864/src/lovely/pytest/docker/compose.py#L212C1-L218C73`
